packages/block-editor/src/components/block-types-list/index.js

```diff
@@ -14,16 +19,30 @@ function BlockTypesList( {
 	onSelect,
 	onHover = () => {},
 	children,
+	label,
 } ) {
+	const composite = useCompositeState();
 	const normalizedItems = includeVariationsInInserterItems( items );
+	const orderId = normalizedItems.reduce(
+		( acc, item ) => acc + '--' + item.id,
+		''
+	);
 
 	return (
 		/*
 		 * Disable reason: The `list` ARIA role is redundant but
 		 * Safari+VoiceOver won't announce the list otherwise.
 		 */
 		/* eslint-disable jsx-a11y/no-redundant-roles */
-		<ul role="list" className="block-editor-block-types-list">
+		<Composite
```

Comment:
Do you think we should expose `Composite` or maybe create a `Listbox` component in `@wordpress/components`? We can live with 10kB extra for now, but I'm sure there is going to be more cases where Reakit can be useful :)